### ISSUE TITLE
Panel serve trips over code in raw cells

### ISSUE THREAD

**Issue by poplarShift** [pyviz/panel#611]
The following notebook cannot be served using `panel serve` while working ok as notebook as such:
(Code cell:)
```
import panel as pn
pn.extension()
app = pn.Row('SPAM')
app.servable()
```
(Raw cell:)
```
some = invalid python code
```

Note that if the raw cell is changed to `foo` it seems to work.

**Comment by jbednar** [comment]
According to [the ipython docs](https://ipython.org/ipython-doc/dev/notebook/nbformat.html#raw-nbconvert-cells), a raw cell is intended for inclusion as-is when nbconvert is used to create output of the type of that raw cell.  Here, if it's a raw Python cell, then that code is ignored by the notebook but is included when output as Python, which is essentially what `panel serve` is doing. It seems valid to me to use this mechanism to pass some additional Python code into the file when Panel converts it to Python, which appears to be happening here, so I'm not sure what the problem is -- if you don't want the raw cell's code to be executed, why have it there?  It all _seems_ to be working as designed, but I may be missing something.

**Comment by poplarShift** [comment]
I wasn't necessarily saying it's a bug but it had me confused for a while. The docs do sound like all `panel serve` cares about is the python code but now I see they launch the "entire" notebook including the raw content.

(I guess I belong to those who (ab)use raw cells as a shorthand for commenting out groups of cells (a single push on `r` is much easier than entering each of them and going ctrl-a ctrl-/) but in hindsight I realize that may not even be common usage.)

**Comment by jbednar** [comment]
Ok, that makes sense.  When you hit `r`, it designates the cell as raw without specifying the type, and [the docs](https://ipython.org/ipython-doc/dev/notebook/nbformat.html#raw-nbconvert-cells) indicate that such text should  be included in *all* nbconvert output.  I'm not sure how useful that convention is, given that few strings will be valid for all nbconvert targets, but that's what it's supposed to do.  The upshot is if that if you use `r` to mark cells not to be parsed by the notebook editor, you're making the notebook unusable by _anything_ based on nbconvert, including Panel. 

We could add a special case to ignore untyped NBConvert raw cells, but that seems like an arbitrary thing to do, and would cause problems for any user who expected untyped raw cells to work as they were designed.  If there are tons of people who see this issue and habitually use `r` in this way, so much that they drown out any people who use it as designed, then we could reconsider, but meanwhile I think you'll have to wean yourself from `r` if you want to to use the notebook anywhere but the native notebook interface. :-)

**Comment by poplarShift** [comment]
No, I'm all for leaving behaviour as is. But maybe the docs could be more specific about how `panel serve` works in this regard? E.g., when I go `jupyter nbconvert --execute`, raw cells get ignored too, so `panel serve`'s behaviour wasn't obvious to me (let alone the fact that `panel serve` is based off `nbconvert`). Though then again it probably makes a lot more sense to someone who's more familiar with the inner workings of notebooks than me :)

**Comment by jlstevens** [comment]
I agree this needs to be explained better in the docs. For completeness, what happens is this:

* `panel serve` is really just a minor extension to `bokeh serve`.
* `bokeh serve` uses `nbconvert.PythonExporter()` ([see here](https://github.com/bokeh/bokeh/blob/cf6c97e35301adb15dd69eb24f4f8af5e75d64e1/bokeh/application/handlers/notebook.py#L120)) . with a preprocessor to strip out magics.

In other words, the behavior is the same as the following, just with magics stripped:

```
jupyter nbconvert notebook.ipynb --to python
```

This means there is no mention of raw cells in either panel or bokeh: the behavior with respect to raw cells is down to how nbconvert turns notebooks into Python code.

**Comment by jbednar** [comment]
Thanks, Jean-Luc.  @poplarShift , we'd be happy to see a PR where this is explained in the docs where it could help users avoid the confusion you had.

**Comment by poplarShift** [comment]
Thanks @jlstevens and @jbednar for the explanations. I summarized the info you gave here and added a little paragraph in the docs.
